price_parser/parser.py

```diff
@@ -36,14 +36,15 @@ def fromstring(cls, price: Optional[str],
         ``price`` string, it could be **preferred** over a value extracted
         from ``currency_hint`` string.
         """
-        amount_text = extract_price_text(price) if price is not None else None
+        currency, source = _extract_currency_symbol(price, currency_hint)
+        amount_text = extract_price_text(price, currency if source == price else None) if price is not None else None
         amount_num = parse_number(amount_text) if amount_text is not None else None
-        currency = extract_currency_symbol(price, currency_hint)
         if currency is not None:
-            currency = currency.strip()
+            currency = currency.group(0).strip()
+            assert isinstance(currency, str)
         return Price(
             amount=amount_num,
-            currency=currency,
+            currency=currency,  # type: ignore
```

Comment:
not overriding Match object with a string may help to remove this `#type: ignore`